src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -673,3 +675,324 @@ The Run handle/object supports few other use-cases.
+- The API ``xrtGraphWait(graphHandle,0)`` is used to wait till the iteration is done. 
+
+     - The API `xrtGraphWait` is used because the host code needs to execute the graph again. 
+- The Graph is executed again for 5 iteration
+- The API ``xrtGraphEnd(graphHandle,0)`` is used to wait till the iteration is done. 
+
+    - After ``xrtGraphEnd`` the same graph should not be executed. 
+
+.. code:: c
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           // run the graph for 3 iteration
+           xrtGraphRun(graphHandle, 3);
+           
+           // Wait till the graph is done 
+           xrtGraphWait(graphHandle,0);  // Use xrtGraphWait if you want to execute the graph again
+           
+           
+           xrtGraphRun(graphHandle,5);
+           xrtGraphEnd(graphHandle,0);  // Use xrtGraphEnd if you are done with the graph execution
+
+
+**Timeout wait scheme**
+
+As shown in the above example ``xrtGraphWait(graphHandle,0)`` performs a busy-wait and suspend the execution till the graph is not done. If desired a timeout version of the wait can be achieved by ``xrtGraphWaitDone`` which can be used to wait for some specified number of milliseconds, and if the graph is not done do something else in the meantime. An example is shown below
+
+.. code:: c++
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           // run the graph for 100 iteration
+           xrtGraphRun(graphHandle, 100);
+           
+            while (1) {
+             auto rval  = xrtGraphWaitDone(graphHandle, 5); 
+              std::cout << "Wait for graph done returns: " << rval << std::endl;
+              if (rval == -ETIME)  {
+                   std::cout << "Timeout, reenter......" << std::endl;
+                   // Do something 
+              }
+              else  // Graph is done, quit the loop
+                  break;
+             }
+
+
+Infinite Graph Execution
+************************
+
+The graph runs infinitely if ``xrtGraphRun`` is called with iteration argument -1. While a graph running infinitely the APIs ``xrtGraphWait``, ``xrtGraphSuspend`` and xrtGraphEnd can be used to stop the graph operation after some number of AIE cycles. The API ``xrtGraphResume`` is used to execute the infinitely running graph again. 
+
+- The API ``xrtGraphRun(graphHandle, -1)`` is used to execute the graph infinitely
+- The API ``xrtGraphWait`` suspends the graph after 3 AIE cycles from the graph starts. 
+
+       - If the graph was already run more than 3 AIE cycles the graph is suspended immediately. 
+- The API ``xrtGraphResume`` is used to restart the suspended graph
+- The API ``xrtGraphSuspend`` is used to suspend the graph immediately
+- The API ``xrtGraphEnd(graphHandle,5)`` is  suspending the graph after 5 AIE cycles from the previous graph start. 
```

Comment:
xrtGraphEnd will end the graph not suspend the graph

Reply:
fixed,